src/request-api.js

```diff
@@ -90,6 +90,10 @@ function requestAPI (path, args, opts, files, buffer, cb) {
     })
   })
 
+  req.on('error', function (err) {
+    return cb(err, null)
+  })
```

Comment:
Following https://nodejs.org/api/http.html#http_http_request_options_callback, an error is only emitted if something with the `request` goes wrong and the callback is only called if we get a response (after the request went well). So, yeah, all good :) 

```
If any error is encountered during the request (be that with DNS resolution, TCP level errors, or actual HTTP parse errors) an 'error' event is emitted on the returned request object.
```